### PR TITLE
SB-21895,SB-21897,SB-21898 Create, update and read api change…

### PR DESCRIPTION
**Pull request by sknirmalkar89**
Support New UserTypes: Student,Administrator,Parent/Guardian
Support for new User Subtypes for the user

### CODE REVIEW

File: actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java [hunk 1]
```diff
@@ -614,8 +614,6 @@ private void validateUserTypeForUpdate(Map<String, Object> userMap, boolean isCu
         ProjectCommonException.throwClientErrorException(
```
Comment (reshmi-nair) [inline actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java:614]: We do not need this teacher validation any more

File: actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java [hunk 2]
```diff
@@ -811,8 +809,6 @@ private void validateUserType(
               ResponseCode.errorTeacherCannotBelongToCustodianOrg.getErrorMessage());
```
Comment (reshmi-nair) [inline actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java:809]: Remove the teacher validation

File: actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java [hunk 3]
```diff
@@ -766,12 +766,15 @@ public void validateFrameworkCategoryValues(
     return frameworkMap.get(key);
   }
 
+  // TODO:  Validate userType with data from form api
   private void validateUserType(Request userRequest) {
     String userType = (String) userRequest.getRequest().get(JsonKey.USER_TYPE);
 
     if (userType != null
```
Comment (reshmi-nair) [inline actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java:773]: Add validation for sub type, if it is not null

File: actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java (hunk 1 above)
Comment (sknirmalkar89) [reply actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java]: ok sure.. Removing the validation

File: actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java (hunk 2 above)
Comment (sknirmalkar89) [reply actors/sunbird-lms-mw/actors/user/src/main/java/org/sunbird/user/actors/UserManagementActor.java]: Removing teacher validation

File: actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java (hunk 3 above)
Comment (sknirmalkar89) [reply actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java:773]: Cannot add currently as we dont have complete subtypes. Once form configuration is done we will change the validation from cache config.

Comment (reshmi-nair) [reply actors/sunbird-lms-mw/actors/sunbird-utils/sunbird-platform-core/common-util/src/main/java/org/sunbird/common/request/UserRequestValidator.java:773]: ok we do this validation during usertype form caching

**Review by reshmi-nair — APPROVED** [review]

**Pull request merged**